### PR TITLE
add fallback image

### PR DESCRIPTION
**Pull request by pldespaigne**
#3120

### CODE REVIEW

File: apps/backend-functions/src/internals/image.ts [hunk 1]
```diff
@@ -75,19 +76,40 @@ async function resize(data: functions.storage.ObjectMetadata) {
...
+        version: 'v2'
+      });
+
+      uploaded.push({ key, url: signedUrl });
+
+    // Fallback : we need to convert the uploaded file into a png
+    // and also generate an access url
+    } else if (key === 'fallback') {
+
+      const pngFileName = basename(resizedImgName, extname(resizedImgName)) + '.png'
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/internals/image.ts:105]: Do you need to add ".png" in the end ? I'm not sure we are doing that for webp. But that's fine for me. Maybe we should.

File: libs/utils/src/lib/media/media.firestore.ts [hunk 2]
```diff
@@ -26,23 +26,23 @@ export function createImgRef(ref: Partial<ImgRef> | string = {}): ImgRef {
...
+      lg: '',
     },
     ..._ref
   };
 }
 
 export function getImgSize(url: string) {
   if (url.includes('avatar')) {
-    return { original: 0, xs: 50, md: 100, lg: 300 };
+    return { original: 0, xs: 50, md: 100, lg: 300, fallback: 0 };
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/media/media.firestore.ts:37]: Hmmm do we need that ?

Comment (pldespaigne) [reply libs/utils/src/lib/media/media.firestore.ts:37]: unfortunately yes, that's how we iterate over the different files we need to create

File: apps/backend-functions/src/internals/image.ts (hunk 1 above)
Comment (pldespaigne) [reply apps/backend-functions/src/internals/image.ts:105]: yes every other image have the `.webp` extension, if we don't do this our png will be named `myPng.webp`

**Pull request merged**